docs/ops/deployment/kubernetes.md

```diff
@@ -262,7 +262,7 @@ spec:
     spec:
       containers:
       - name: jobmanager
-        image: flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest{% endif %}
+        image: flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest # The 'latest' tag contains the latest released version of Flink for a specific Scala version which will mismatch with your application over time.{% endif %}
```

Comment:
```suggestion
        image: flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest # The 'latest' tag contains the latest released version of Flink for a specific Scala version which can mismatch with the Scala version of your application.{% endif %}
```
Will the Scala version definitely mismatch?
I am wondering why this comment should be in all places. This is for dev docs.
Could we not put this comment to that note in the relevant docker/tags chapter?

Reply:
In my experience: Yes, the versions will mismatch at some point. 
Also with a Scala mismatch the problem is big, yet with a Flink version mismatch I suspect more subtle problems to occur.

Also I consider the choice of having Scala 2.12 the version used for "latest" a choice for today which may change (As I mentioned in the mailing list I am actually against having a "latest" for something like Flink).

Note that this comment is ONLY shown when the documentation is generated for a non-stable "SNAPSHOT" build of the documentation. So the normal documentation should not show this.
I duplicated it in all of those places because these are Kubernetes config files I expect to be used by operations people who will have limited knowledge of the application at hand.

Reply:
Alright but if we and user always use the same Scala version, it will never mismatch, right?

Reply:
For me one big trigger was that I had a mis match and the errors were not helping me figure out what was happening.

Reply:
This sounds fine for me to fix.
It just sounds now as if it `will` definitely mismatch, although it looks like it `can` mismatch if not aligned with the image.

Reply:
How about this:
```
flink:latest # The 'latest' tag contains the latest released version of Flink for a specific Scala version which can conflict with the versions, used by your application.
```
Somehow this sounds confusing:
`which will mismatch with your application over time`, in particular `over time`.

Reply:
I think we should be way more explicit about this; scala users should _never_ use `latest`; it is simply to easy to run into issues. If we release a newer release, with scala 2.13 support we suddenly break everything.
I would explicitly phrase it like that; use it only if you are only using java, otherwise select a specific Flink version with a matching scala version.

We should also look into providing a `latest_<scala.version>` tag to make this easier for scala users.

Reply:
We added a [note](https://github.com/apache/flink/pull/11245/files#diff-a530b60989203f9e1f03c64c57deb56cR49) for this which can be promoted to warning.
I think the idea here is to keep the comment in this particular line (and others like this) brief enough as it is a code example.